### DISCUSSION TITLE
new powerdns server setup

### DISCUSSION THREAD

**Discussion by hpcmtint** [PowerDNS/pdns#13754] [Q&A]
We're transitioning from a commercial DDI solution to an open-source stack, incorporating tools like PowerDNS, phpIPAM, and Kea DHCP to handle our DDI needs.

As I'm relatively inexperienced with DNS server configuration, I'm seeking guidance on setting up PowerDNS for resolving both internal and external domain names. Specifically, I'm curious if it's necessary to implement both authoritative and recursive DNS servers for this purpose. Additionally, I aim to configure and manage these components using Docker Compose.

Could anyone provide some advice or insights on this setup?

**Comment by micheloe** [comment]
This is a rather late response to your question, but I'm afraid more details are needed to provide some guidance. You speak about "internal and external domain names", but you don't specify if you are authoritative for these domain names. Also, internal and external can mean different things, but I read it as private (internal) and public (external) domain names. I assume you are authoritative for both and want to run these on PowerDNS.

In that case several PowerDNS auth instances can provide this functionality. You can choose to run the private / public instances on separate machines, or run an (i.e. private) instance on an alternative port. This sort of avoids leaking internal / private domains to the public.

For systems that need to resolve internal domains, you'd use recursor(s) to direct queries to internal domains to the internal / private auth servers or naturally recurse to external / public domains otherwise. You can also use dnsdist for this, depending on your situation.
